### PR TITLE
OZ-557: Add documentation for data flows between OpenMRS and ERPNext.

### PR DESCRIPTION
**Pull request by corneliouzbett**
Issue: https://mekomsolutions.atlassian.net/jira/software/c/projects/OZ/issues/OZ-557

This PR adds documentation for the data flows between OpenMRS and ERPNext. It includes the main flows, prerequisite flows, and options for each data flow. The data flows documented are:

- OpenMRS Patient → ERPNext Customer
- OpenMRS Billable Items ⭆ ERPNext Quotation
- OpenMRS Billable Item → ERPNext Quotation Line
- OpenMRS Visit → ERPNext Quotation

### CODE REVIEW

File: docs/users/emr-erp.md [hunk 1]
```diff
@@ -62,6 +62,106 @@ flowchart LR
...
+    - File:<br/>`ozone/distro/configs/eip-erpnext-openmrs/application.properties`
+    - Property name:<br/>`erpnext.openmrs.enable.patient.sync`
+    - Possible values:<br/>`false` (<small>**default**</small>), `true`
+
+### OpenMRS Billable Items ⭆ ERPNext Quotation
+
+#### Summary
+As soon as the first billable item is ordered for a patient in OpenMRS,
+a quotation is created in ERPNext for the corresponding customer.
+Billable items ordered in the same visit are added to the same quotation as quotation items.
```
Comment (mks-d) [inline docs/users/emr-erp.md:104]: >As soon as the first billable item is ordered for a patient in OpenMRS,
a quotation is created in ERPNext for the corresponding customer that is associated with the OpenMRS patient's visit.
All billable items ordered for a patient within the same OpenMRS visit are added to the ERPNext quotation associated with this OpenMRS visit.

File: docs/users/emr-erp.md [hunk 2]
```diff
@@ -62,6 +62,106 @@ flowchart LR
...
+
+#### Main Flow
+``` mermaid
+flowchart LR
+    a["OpenMRS visit"]-- 1-to-1 -->b["ERPNext quotation"]
+```
+
+#### Prerequisite Flows
+
+- [OpenMRS patient → ERPNext customer](#openmrs-patient-erpnext-customer)
```
Comment (mks-d) [inline docs/users/emr-erp.md:163]: This one is also a prerequisite flow here:

>OpenMRS Billable Items ⭆ ERPNext Quotation

File: docs/users/emr-erp.md [hunk 3]
```diff
@@ -62,6 +62,106 @@ flowchart LR
...
+
+None
+
+### OpenMRS Visit → ERPNext Quotation
+
+#### Summary
+As soon as the first billable item is ordered for a patient in OpenMRS,
+a draft quotation is created in ERPNext based on the patient's visit.
+When the visit is ended, the quotation is submitted.
+Following submission, the quotation is eligible for conversion into a sales order.
```
Comment (mks-d) [inline docs/users/emr-erp.md:153]: Let's stick to defining the flow, in which case I believe that it is simply the following:
>Ending a patient's visit in OpenMRS submits the ERPNext quotation associated with this visit. 

This is standard ERPNext stuff right? In which case we don't need to say it.
>Following submission, the quotation is eligible for conversion into a sales order.

File: docs/users/emr-erp.md [hunk 4]
```diff
@@ -62,6 +62,106 @@ flowchart LR
...
+=== "Behaviour"
+
+    - <small>**default**</small> &nbsp; An OpenMRS patient is synchronised as an ERPNext customer when a first billable item is ordered from OpenMRS.
+    - <small>_optional_</small> &nbsp; An OpenMRS patient is always synchronised as an ERPNext customer.
+
+=== "Configuration"
+
+    - File:<br/>`ozone/distro/configs/eip-erpnext-openmrs/application.properties`
+    - Property name:<br/>`erpnext.openmrs.enable.patient.sync`
+    - Possible values:<br/>`false` (<small>**default**</small>), `true`
```
Comment (mks-d) [inline docs/users/emr-erp.md:97]: Just fyi, my take at trying to documents a flow's options. Let me know what you think.

**Review by mks-d — CHANGES_REQUESTED** [review]
Great @corneliouzbett, very much in line with the standard way that I have tried to put in place for describing Ozone's data flows. A couple of changes requested anyway.

File: docs/users/emr-erp.md (hunk 4 above)
Comment (corneliouzbett) [reply docs/users/emr-erp.md:97]: Unlike the table format, this is indeed well-organized and compact, reducing unnecessary white space. It presents all the necessary information clearly and concisely. Good job🥇!

File: docs/users/emr-erp.md (hunk 3 above)
Comment (corneliouzbett) [reply docs/users/emr-erp.md:153]: Agreed.

**Pull request merged**